core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -137,6 +138,22 @@ trait EntityDecoderInstances {
     EntityDecoder.decodeBy(MediaRange.`*/*`)(collectBinary)
   }
 
+  implicit def sourceDecoder[A](implicit W: EntityDecoder[A]): EntityDecoder[Process[Task,A]] =
+    new EntityDecoder[Process[Task,A]] {
+      override def decode(msg: Message): DecodeResult[Process[Task,A]] = {
+        val result = msg.body.flatMap{ byteVector =>
+          val fakeMessage = Request(body = Process.emit(byteVector))
```

Comment:
Do we know if there is any other HTTP client that exposes chunked encoding as a stream?